src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/ServiceRecipientsSectionController.cs

```diff
@@ -1,57 +1,76 @@
-                        }
-                    }
-                };
-            }
-
-            return Ok(model);
-        }
-
-        [HttpPut]
-        [Authorize(Policy = PolicyName.CanManageOrders)]
-        public ActionResult Update(string orderId, ServiceRecipientsModel model)
-        {
-            _cannedData[orderId] = model ?? throw new ArgumentNullException(nameof(model));
-
-            return NoContent();
-        }
-
-        private static readonly Dictionary<string, ServiceRecipientsModel> _cannedData = new Dictionary<string, ServiceRecipientsModel>();
-    }
-}
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using NHSD.BuyingCatalogue.Ordering.Api.Extensions;
+using NHSD.BuyingCatalogue.Ordering.Api.Models;
+using NHSD.BuyingCatalogue.Ordering.Application.Persistence;
+using NHSD.BuyingCatalogue.Ordering.Common.Constants;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.Controllers
+{
+    [Route("api/v1/orders/{orderId}/sections/service-recipients")]
+    [ApiController]
+    [Produces(MediaTypeNames.Application.Json)]
+    [Authorize(Policy = PolicyName.CanAccessOrders)]
+    public sealed class ServiceRecipientsSectionController : ControllerBase
+    {
+        private readonly IOrderRepository _orderRepository;
+        private readonly IServiceRecipientRepository _serviceRecipientRepository;
+
+        public ServiceRecipientsSectionController(IOrderRepository orderRepository,
+            IServiceRecipientRepository serviceRecipientRepository)
+        {
+            _orderRepository = orderRepository ?? throw new ArgumentNullException(nameof(orderRepository));
+            _serviceRecipientRepository = serviceRecipientRepository ??
+                                          throw new ArgumentNullException(nameof(serviceRecipientRepository));
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<ServiceRecipientsModel>> GetAllAsync(string orderId)
+        {
+            var order = await _orderRepository.GetOrderByIdAsync(orderId);
+
+            if (order is null)
+            {
+                return NotFound();
+            }
+
+            var serviceRecipients = (await _serviceRecipientRepository.ListServiceRecipientsByOrderId(orderId)).ToList();
+
+            var primaryOrganisationId = User.GetPrimaryOrganisationId();
+
+            if (primaryOrganisationId != order.OrganisationId)
+            {
+                return Forbid();
+            }
+
+            var recipientModelList = serviceRecipients.Select(recipient => new ServiceRecipientModel
+            {
+                OdsCode = recipient.OdsCode,
+                Name = recipient.Name
+            }).ToList();
+
+            var model = new ServiceRecipientsModel
+            {
+                ServiceRecipients = recipientModelList
+            };
+
+            return model;

```

Comment:
Any reason to not return an OK(model)? Just thinking for consistency reasons.

Reply:
This was in response to one of @bjss-chris previous PR reviews, stating that you dont actually need to return Ok(model), but by returning model it still returns a 200. I can change it back if you prefer, but it is also in other areas of the code

Reply:
@sam-roberts-bjss there's no benefit to returning `Ok(model)` – it just adds complexity to the code and tests that isn't needed and the result is exactly the same. However, it relies on people making sure they are using the `ActionResult<T>` return type rather than `ActionResult` or `IActionResult` but we should be doing this anyway for the reasons outlined [here](https://docs.microsoft.com/en-us/aspnet/core/web-api/action-return-types?view=aspnetcore-3.1#actionresultt-type) by MS.

We can always discuss this afternoon but my recommendation would be for @RuthBerry96 to leave this as is.